src/server/pps/server/api_server.go

```diff
@@ -2845,34 +2845,14 @@ func (a *apiServer) RunCron(ctx context.Context, request *pps.RunCronRequest) (r
 
 	cron := pipelineInfo.Input.Cron
 
-	var latestTime time.Time
-	files, err := pachClient.ListFile(cron.Repo, "master", "")
-	if err != nil && !pfsServer.IsNoHeadErr(err) {
-		return nil, err
-	} else if err != nil || len(files) == 0 {
-		// File not found, this happens the first time the pipeline is run
-		latestTime, err = types.TimestampFromProto(cron.Start)
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		// Take the name of the most recent file as the latest timestamp
-		// ListFile returns the files in lexicographical order, and the RFC3339 format goes
-		// from largest unit of time to smallest, so the most recent file will be the last one
-		latestTime, err = time.Parse(time.RFC3339, path.Base(files[len(files)-1].File.Path))
-		if err != nil {
-			return nil, err
-		}
-	}
-
 	// We need the DeleteFile and the PutFile to happen in the same commit
 	_, err = pachClient.StartCommit(cron.Repo, "master")
 	if err != nil {
 		return nil, err
 	}
 	if cron.Overwrite {
-		// If we want to "overwrite" the file, we need to delete the file with the previous time
-		err := pachClient.DeleteFile(cron.Repo, "master", latestTime.Format(time.RFC3339))
+		// get rid of any files, so the new file "overwrites" previous runs
+		err = pachClient.DeleteFile(cron.Repo, "master", "")
```

Comment:
Nice, this is a much more reliable way to overwrite.

This does implement a slight change in semantics right? Previously if you had a pipeline with a non-overwrite cron and then you updated the pipeline to an overwrite cron, only the most recent tick would get overwritten, now the entire thing does. The new semantics seem much better to me, although it's an admittedly rare case. This is mostly just a sanity check to make sure I'm understanding correctly, no request for change here.

Reply:
Yeah, that's true. My guess is that this is inline with typical expectations

And thanks!